### ISSUE TITLE
[spinner @alpha 9] AoT compilation causes high CPU use via requestAnimationFrame()

### ISSUE THREAD

**Issue by RoxKilly** [angular/material2#1570]
#### Bug, feature request, or proposal:

Bug
#### What is the expected behavior?
- `md-spinner` should not use more CPU when app is AoT compiled than when it is JiT compiled
- `md-spinner` should not use CPU resources once it is taken out of the DOM
#### What is the current behavior?

App uses AoT compilation (JiT is not affected) and has `<md-spinner>` in the template
- High CPU usage even once the spinner is removed from the DOM
- `Zonetask.invoke` calls [`window.requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) in an infinite loop, choking the CPU
- Minimizing the browser brings CPU utilization down. Restoring does the opposite
#### What are the steps to reproduce?

**app.component.html**

```
<md-card-content *ngIf="loading">
    <md-spinner></md-spinner>
</md-card-content>
<md-card-content *ngIf="!loading">
    Content goes here
</md-card-content>
```

`loading===true` when `AppComponent` loads, so the spinner is shown. In `ngOnInit`, after data fetched from the server is received, `loading=false`.  The spinner disappears but CPU utilization remains high.  This only affects AoT app (not in-browser compiled JiT app).

I replaced

```
<md-spinner></md-spinner>
```

with...

```
<md-progress-bar mode="indeterminate"></md-progress-bar>
```

and I no longer experience the problem.
#### What is the use-case or motivation for changing an existing behavior?

This makes `md-spinner` too costly to use with `AoT compilation`
#### Which versions of Angular, Material, OS, browsers are affected?

Tested on `Angular 2.0.1`, `@angular/compiler-cli 0.6.4`, `Material 9-3`, `Firefox 49`, `Chrome 54`, `Windows 7 x64`
#### Is there anything else we should know?

Get more details [from this issue](https://github.com/angular/angular/issues/12253#issuecomment-254139552)

**Comment by ivoviz** [comment]
@crisbeto @RoxKilly The PR might just solve the problem I reported long ago (#1283) - just saying, since it's not referenced in the PR itself, but probably can be closed after it's merged.

**Comment by crisbeto** [comment]
Yeah, let's see, I didn't mark it as "fixes" on purpose since it may not fix it completely.

**Comment by jelbourn** [comment]
@RoxKilly an experimental release for alpha.10 with this fix is on npm under the `@experimental` tag- does the change improve what you had been seeing?

**Comment by RoxKilly** [comment]
@jelbourn `alpha 10 experimental-yogurt` seems to have fixed the issue. CPU and memory consumption are both down

**Comment by jelbourn** [comment]
Sounds good- I'll make this as closed by #1635 then which will be in the alpha.10
